lib/README.md

```diff
@@ -1,145 +1,181 @@
 
-For features that may be merged to Core, it's best to use a `wp_` prefix for functions or a `WP_` prefix for classes.
+Files at the root of `/lib` are generally considered to contain "evergreen" code. Such code is both fundamental to the proper functioning of the plugin, and also so often updated that versioning it between WordPress releases is not practical. Changes to these files are merged into Core as required.
 
-This applies to both experimental and stable features.
+## Best practices
+
+There are a few best practices that should be observed when adding new files to the Gutenberg plugin.
 
-Using the `wp_` prefix avoids us having to rename functions and classes from `gutenberg_` to `wp_` if the feature is merged to Core.
+### Using `gutenberg` suffixes/prefixes vs. `wp` prefixes
 
-Functions that are intended solely for the plugin, e.g., plugin infrastructure, should use the `gutenberg_` prefix.
+To avoid naming conflicts with WordPress Core and other plugins, the Gutenberg plugin uses the `gutenberg` identifier in many of its PHP classes and function names, e.g., `WP_Theme_JSON_Gutenberg` and `gutenberg_get_block_editor_settings`.
 
-#### Feature that might be merged to Core
+This is especially so for classes and functions whose functionality is ubiquitous and constantly being updated — so-called "evergreen" code. Anything related to `WP_Theme_JSON_Gutenberg` is a good example of this: this class controls the way Gutenberg processes and outputs global styles and much more, and its methods are called in many places. In every aspect of plugin functionality, we want plugin users to have access to the latest versions of these files, even if they are running an older version of WordPress.
 
 ```php
-function wp_get_navigation( $slug ) { ... }
+/**
+* Returns something useful.
+*
+* @since 6.2.0 Updates to something even more useful.
+* @since 6.3.0 Now more useful than ever.
+*
+* @return string Something useful.
+*/
+function gutenberg_get_something_useful() {
+	// ...
+}
 ```
 
-#### Plugin infrastructure that will never be merged to Core
+When porting new functions into Core, the function must be renamed to use the `wp_` prefix for functions or a `WP_` prefix for classes.
 
 ```php
-function gutenberg_get_navigation( $slug ) { ... }
+/**
+* Returns something useful.
+*
+* @since 6.2.0 Updates to something even more useful.
+* @since 6.3.0 Now more useful than ever.
+*
+* @return string Something useful.
+*/
+function wp_get_something_useful() {
+	// ...
+}
 ```
 
-### Group PHP code by _feature_
+Plugin code that is stable and expected to be merged "as-is" into Core in the near future can use the `wp_` prefix for functions or a `WP_` prefix for classes.
 
-Developers should organize PHP into files or folders by _feature_, not by _component_.
+The caveat is that no functions or classes with the same names already exist in Core. A quick codebase search will also help you know if your new names are unique.
 
```

Comment:
The wording here feels slightly awkward to me. Rather than `caveat`, can we be explicit that there must not be a duplicate? Perhaps something like:

> When doing so, care must be taken to ensure that no duplicate declarations to create functions or classes exist between Gutenberg and WordPress core code. A quick codebase search will also help you know if your new names are unique.

I think that should bridge the sentence above and the sentence below? Feel free to disregard this comment, though!

Reply:
Thank you for proof reading this. I have shamelessly plagiarized your example sentence. Much better. 🙇🏻